services/s3control/src/main/java/software/amazon/awssdk/services/s3control/internal/interceptors/EndpointAddressInterceptor.java

```diff
@@ -81,7 +86,7 @@ private String resolveHost(SdkHttpRequest request, String accountId, S3ControlCo
 
         }
         validateComponentIsHostnameCompliant(accountId, "account id");
-        return String.format("%s.%s", accountId, host);
+        return hasEndpointOverride ? host : String.format("%s.%s", accountId, host);
```

Comment:
New changes look good, but looks like this still needs to be reverted along with the accompanying test.